### PR TITLE
sceneNum -> sceneId

### PR DESCRIPTION
**Pull request by hensldm**
This includes the play struct member (sceneNum -> sceneId), the savecontext struct member (savedSceneNum -> savedSceneId), as well as other instances of it around.

### CODE REVIEW

**Comment by fig02** [comment]
Thanks for this.

Some more stuff that needs to be changed:
`HorseData`, `HorseSpawn`, `struct_8011F9B8`, `EnHorseSpawnpoint`, `EntranceInfo`, `RestrictionFlags`, temp in `Interface_SetSceneRestrictions`

play.loadedScene can possibly be `curSceneTableEntry`? not exactly in scope of this pr but something to think about

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c [hunk 1]
```diff
@@ -243,7 +243,7 @@ void DoorShutter_Init(Actor* thisx, PlayState* play2) {
         BossDoorInfo* phi_v1_2;
 
         for (phi_v1_2 = &D_80998288[0], i = 0; i < ARRAY_COUNT(D_80998288) - 1; i++, phi_v1_2++) {
-            if (play->sceneNum == phi_v1_2->dungeonScene || play->sceneNum == phi_v1_2->bossScene) {
+            if (play->sceneId == phi_v1_2->dungeonScene || play->sceneId == phi_v1_2->bossScene) {
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c:246]:
```suggestion
            if (play->sceneId == phi_v1_2->dungeonSceneId || play->sceneId == phi_v1_2->bossSceneId) {
```

File: src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c [hunk 2]
```diff
@@ -250,7 +250,7 @@ void ObjOshihiki_SetColor(ObjOshihiki* this, PlayState* play) {
     paramsColorIdx = (this->dyna.actor.params >> 6) & 3;
 
     for (i = 0; i < ARRAY_COUNT(sScenes); i++) {
-        if (sScenes[i] == play->sceneNum) {
+        if (sScenes[i] == play->sceneId) {
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c:253]:
```suggestion
        if (sSceneIds[i] == play->sceneId) {
```

File: src/overlays/gamestates/ovl_select/z_select.c [hunk 3]
```diff
@@ -519,14 +519,14 @@ void MapSelect_PrintMenu(MapSelectState* this, GfxPrint* printer) {
     for (i = 0; i < 20; i++) {
         GfxPrint_SetPos(printer, 9, i + 4);
 
-        scene = (this->topDisplayedScene + i + this->count) % this->count;
-        if (scene == this->currentScene) {
+        sceneId = (this->topDisplayedScene + i + this->count) % this->count;
+        if (sceneId == this->currentScene) {
```
Comment (Dragorn421) [inline src/overlays/gamestates/ovl_select/z_select.c:523]: these are actually not scene ids, just indices into z_select's `sScenes`

(probably should be renamed to "map" or "entrances" as a whole eventually at a later time)

Comment (hensldm) [reply src/overlays/gamestates/ovl_select/z_select.c]: Sounds good. Just reverted this file for the time being.

**Review by Dragorn421 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**